platform/osx/crash_handler_osx.mm

```diff
@@ -135,7 +135,11 @@ static void handle_crash(int sig) {
 				args.push_back("-o");
 				args.push_back(_execpath);
 				args.push_back("-arch");
+#if defined(__x86_64) || defined(__x86_64__) || defined(__amd64__)
 				args.push_back("x86_64");
+#elif defined(__aarch64__)
+				args.push_back("arm64");
```

Comment:
That shouldn't happen, but in theory of new arches were added then the call would become invalid as there's no `#else` condition. Maybe it should raise an error in `#else`?

Or is it mandatory to set the arch, or can we get away with just letting it infer things?

Reply:
Moved `args.push_back("-arch");` into the ifdef, it's an optional flag. This way it should try to infer architecture if we do not know it.